### ISSUE TITLE
pd.concat with names=None

### ISSUE THREAD

**Issue by amgcc** [pandas-dev/pandas-stubs#883]
**Describe the bug**
`pd.concat` should accept `names=None` as it is the default value.

**To Reproduce**
1. Provide a minimal runnable `pandas` example that is not properly checked by the stubs.

```
def maybe_concat(
    dd: dict[str, pd.DataFrame], names: list[str] | None = None
) -> pd.DataFrame:
    """Concatenate a dict of dataframes if available."""
    return pd.concat(dd, names=names) if dd else pd.DataFrame()
```

2. Indicate which type checker you are using (`mypy` or  `pyright`).
**mypy**
3. Show the error message received from that type checker while checking your example.
```
error: Argument "names" to "concat" has incompatible type "list[str] | None"; expected "list[str]"  [arg-type]
```

**Please complete the following information:**
 - OS: [e.g. Windows, Linux, MacOS] **Linux**
 - OS Version [e.g. 22] **Fedora 7**
 - python version **3.11.8**
 - version of type checker **1.8.0**
 - version of installed `pandas-stubs` **2.1.4**

**Comment by Dr-Irv** [comment]
Yes, that's an error.  Easy fix to add the `| None` to the `names` argument in `pandas-stubs/core/reshape/concat.pyi`  PR with tests welcome.

**Comment by amgcc** [comment]
Ill try and tackle this.

**Comment by Aftabby** [comment]
Hello, @Dr-Irv 
I'm assigning the issue to myself, as it seems no one else has done any improvement on it yet. take

**Comment by Aftabby** [comment]
take

**Comment by Aftabby** [comment]
Hello @Dr-Irv ,
Will you kindly check if issue has been solved with #894 or not?
